lib/screen/settings/crypto/wallet_detail/wallet_detail_page.dart

```diff
@@ -114,15 +96,29 @@ class _WalletDetailPageState extends State<WalletDetailPage> with RouteAware {
 
   @override
   void didPopNext() {
+    _callBlocWallet();
+    _callFetchConnections();
+  }
+
+  void _callBlocWallet() {
     final cryptoType = widget.payload.type;
-    final address = widget.payload.walletAddress.address;
-    context
-        .read<WalletDetailBloc>()
-        .add(WalletDetailBalanceEvent(cryptoType, address));
+    switch (cryptoType) {
+      case CryptoType.ETH:
+      case CryptoType.XTZ:
+        context
+            .read<WalletDetailBloc>()
+            .add(WalletDetailBalanceEvent(cryptoType, address));
+        break;
+      case CryptoType.USDC:
+        context.read<USDCBloc>().add(GetUSDCBalanceWithAddressEvent(address));
+        break;
+      default:
+        // do nothing
+        break;
+    }
     if (cryptoType == CryptoType.ETH) {
       context.read<USDCBloc>().add(GetUSDCBalanceWithAddressEvent(address));
```

Comment:
why need to keep both if & switch here?

Reply:
yeah, i was thinking about merging it too. The reason i leave it that way is to separate purpose of the bloc calling. in the SWITCH, it is for the balance on the top (main crypto).
the IF is for balance of layer 'n' token inside the main crypto.

Reply:
now i found out there is a better way is to use continue